kernelci/build.py

```diff
@@ -538,12 +538,21 @@ def get(self, *keys):
                 break
         return value
 
-    def add_step(self, data):
-        """Add meta-data for a build step
+    def update_step(self, data):
+        """Update meta-data for a build step
+
+        Update meta-data for a build step so that if there is already a step
+        with the same name, the new data overwrites the old one, otherwise the
+        data is appended at the end.
 
         *data* is the data for the step, following the schema
         """
-        self._steps.append(data)
+        for i, item in enumerate(self._steps):
+            if item['name'] == data['name']:
+                self._steps[i] = data
+                break
```

Comment:
The for-loop provides references to the dictionaries in the list, so it's also possible to replace their content directly rather than using indexes:
```suggestion
        for step in self._steps:
            if step['name'] == data['name']:
                step.clear()
                step.update(data)
                break
```
I find it a bit cleaner to solely rely on the iterator, but I guess using the indexes also works.